### ISSUE TITLE
Doesn't download default (creator published) subtitles.

### ISSUE THREAD

**Issue by FrostedVolcano** [adithya-s-sekhar/advanced-youtube-client-ayc#8]
**Description:** When enabling subtitles (both .srt and embedded .vtt types), if the video on YouTube has both "auto-generated" and "Default" then only "Default" or both of these subtitles should be downloaded/added to the video file. But it only downloads/adds the "auto-generated" subs. "auto-generated" is the subtitles generated by YouTube and "Default" is the subtitle published by the creator. I also tried editing the "envetup.bat" , where I tried adding `--write-subs` in various ways, but it only downloads the auto generated one. I also tried removing `--write-auto-subs` and adding `--write-subs` but that doesn't download any subs. 
**Issue:** If the video has default subtitiles, then the default or both default and auto-generated should be downloaded, but it only downloads auto-generated subs.
**Expected results:** Either only default or both default and auto generated subs should be downloaded.

**Comment by adithya-s-sekhar** [comment]
Hi can you share one of these video links? I've been wanting to test this behaviour for some time.

**Comment by adithya-s-sekhar** [comment]
@FrostedVolcano So I have tried [this](https://www.youtube.com/watch?v=MV1qaFv4VUg) URL which has both a creator published eng subtitle and an auto generated eng subtitle. It downloaded the creator published subtitle when you set it to write the subtitle file externally. But it prefers the auto generated one when you try to embed it.

This is something currently being worked on by the yt-dlp team here: https://github.com/orgs/yt-dlp/projects/4/views/1?pane=issue&itemId=14786617

For now, we can fix this by writing to disk first, embed, then delete the creator published subtitle with `--write-subs --write-auto-subs --embed-subs --compat-options no-keep-subs`.

You can try this updated envsetup.bat file, see if it fixes your issue.
[envsetup.zip](https://github.com/user-attachments/files/16104390/envsetup.zip)

Or this test build on telegram, haven't setup cd on github yet. 
[ayc-v4.xx.zip](https://t.me/ayc_news/138)

**Comment by FrostedVolcano** [comment]
@adithya-s-sekhar sorry, I couldn't reply to your first comment in time, I don't check GitHub frequently. 

Today, I deleted my old ayc folder and downloaded v4.22 again. I tried [this video](https://youtu.be/Tbg2XJPNFUc), it has both default (creator published) and auto-generated subs. I tried this video before and I am trying it now after checking your comment. I tried both save externally as .srt and embeded as .vtt. Both downloaded the auto-generated ones. 

Another thing, today morning ayc was not downloading any videos, I tried yt-dlp cli, which did download videos but it showed an warning: 
`WARNING: [youtube] Tbg2XJPNFUc: Native nsig extraction failed: Trying with PhantomJS`
`n = 5jaiZYPkNeTMqyf2M ; player = https://www.youtube.com/s/player/b22ef6e7/player_ias.vflset/en_US/base.js`
So I downloaded PhantomJS, put the .exe in the bin folder and it started downaloading normally. I don't know whether PhantomJS fixed it or updating yt-dlp or reinstalling ayc.
This is off topic, but I thought this information might help you.

Finally, I would like to thank you a lot for such great effort. You are spending your time listening to random people on the internet. And also for making such a great tool and making it open source. God bless you.

**Comment by adithya-s-sekhar** [comment]
@FrostedVolcano Thank you for your kind words, this was something I made for myself a long time ago to simplify youtube-dl. I also use this quite a lot. I tried your URL, it has the issue as well. This needs some more experimenting.

**Comment by adithya-s-sekhar** [comment]
Fixed in [v4.23](https://github.com/adithya-s-sekhar/advanced-youtube-client-ayc/releases).
I have added a toggle to disable auto generated subtitles it should fix your issue. In subtitle settings, disable auto generated subtitles and change subtitle languages to all. This would download only the creator published subtitles.

Sometimes the creators won't label their languages properly or youtube could add random characters at the end of the name. So setting language to all will be fine.
